internal/controller/nnf_node_block_storage_controller.go

```diff
@@ -288,6 +292,32 @@ func (r *NnfNodeBlockStorageReconciler) Reconcile(ctx context.Context, req ctrl.
 
 	nodeBlockStorage.Status.Ready = true
 
+	log.V(1).Info("Reconcile reached fence check point")
+
+	// Check for fence requests after all normal BlockStorage operations are complete
+	// This ensures the storage is properly configured before attempting fence operations
```

Comment:
Since ProcessFenceRequests() lists all NnfNodeBlockStorages on the Rabbit, I don't think we can guarantee that the storage is fully set up. Two workflows using different compute nodes on the same Rabbit will both be capable of triggering ProcessFenceRequests(). Whenever that happens it will try to rip the StorageGroups out from under the other workflows NnfNodeBlockStorage.